rpc/src/rpc/native/txn.rs

```diff
@@ -0,0 +1,283 @@
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+
+use anyhow::{anyhow, Result};
+use ethers::{
+    prelude::*,
+    types::transaction::eip2930::AccessList,
+    utils::{keccak256, rlp},
+};
+use futures::stream::{FuturesOrdered, TryStreamExt};
+use tokio::sync::Mutex;
+use trace_decoder::trace_protocol::ContractCodeUsage;
+use trace_decoder::trace_protocol::TxnTrace;
+use trace_decoder::trace_protocol::{TxnInfo, TxnMeta};
+
+use super::tracing_options_diff;
+use super::{tracing_options, Block, Http};
+
+/// Processes the transactions in the given block and updates the code db.
+pub(super) async fn process_transactions(
+    block: &Block<H256>,
+    provider: Arc<Provider<RetryClient<Http>>>,
+    code_db: Arc<Mutex<HashMap<H256, Vec<u8>>>>,
+) -> Result<Vec<TxnInfo>> {
+    let mut futures_ordered = FuturesOrdered::new();
+
+    for tx_hash in &block.transactions {
```

Comment:
Last time we tested blocks, I recall we needed to reverse the order here. Has this been tested since @frisitano?

Reply:
Yes, I tested this on 200 blocks with a 99.5% success rate. The change was made below on line 30 in which we `push_back` instead of the prior `push_front`, see [ref](https://github.com/fractal-zkp/zero-bin/blob/5234bc9dc291750aa51b3090379acb9160fc57cf/rpc/src/rpc/native/txn.rs#L27-L33).

Reply:
Awesome, thanks!
The failure being due to the branch collapsing bug mentioned in the related issue I presume?

Reply:
No that issue has been fixed (hacked) by fetching proofs after block execution and inserting short nodes (extension and leafs) into the trie builder. 

This failure is due to an error when trying to fetch a proof for an account - see https://github.com/0xPolygonZero/zero-bin/pull/81#issuecomment-2133679551. I haven't had an opportunity to look into it.